enhancements/windows-containers/windows-machine-config-operator.md

```diff
@@ -0,0 +1,185 @@
+secret with a public key that corresponds to the private key. In addition it
+will have all the pre-requisite steps needed for setting the VM up for key based
+SSH connections.
+
+When the cluster admin wants to bring up Windows worker nodes, they will create
+a [MachineSet](https://github.com/openshift/machine-api-operator/blob/master/config/machineset.yaml)
+where they reference a Windows OS image which has the Docker container runtime
+add-on enabled. They will also specify the user data secret created by WMCO as
+described above. WMCO will watch for Machine objects with a special label that
+indicate that they map to Windows VMs. Once the Machine object is in the
+["provisioned" phase](https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/machine-instance-lifecycle.md),
+WMCO will connect to the associated VM to transfer the binaries and perform the
+following steps for it to join the cluster as a worker node:
+* Transfer the required binaries
+* Configure the kubelet service using WMCB
+* Run the hybrid-overlay-node.exe binary
+* Configure the kubelet service for CNI using WMCB
+* Configure kube-proxy
+
+When the cluster admin wants to scale up or down the number of Windows workers,
+they will edit the (replicas) https://github.com/openshift/machine-api-operator/blob/master/config/machineset.yaml#L11
+field in the Windows Machine Set spec.
+
+### User Stories
+
+The user stories are collected using the links below:
+* (Windows Containers Productization Epic) https://issues.redhat.com/browse/WINC-182
+* (Machine API migration)[https://issues.redhat.com/issues/?filter=12347681]
+* (Filter out "worker" windows node from being tagged by the MCC)[https://issues.redhat.com/browse/GRPA-2213]
+
+### Risks and Mitigations
+
+* The main risk is that OVN Hybrid networking will not GA in 4.6. In that
+  scenario a third-party networking solution that has been validated for all the
+  targeted platforms will need to be explored.
+* Linux workloads that have blanket tolerations has the potential to land on a
+  Windows node and fail. The mitigation for this is to document this clearly in
+  the official Windows Containers documentations. Other OpenShift operator
+  developers should also ensure that they are not using blanket toleration.
+
+### Test Plan
+
+The WMCO repository has already been integrated with Prow CI. We are running
+e2e tests for every PR that is opened. These e2e tests involve bringing up a
+cluster on all supported cloud providers, instantiating Windows nodes and
+running the following tests:
+* Basic node validation
+* East west networking between Linux and Windows workloads
+* North south networking
+
+### Graduation Criteria
+
+This enhancement will start as GA.
+
+### Upgrade / Downgrade Strategy
+
+This is an involved topic and we will be opening a separate enhancement
+detailing it.
+
+### Version Skew Strategy
+
+We plan to maintain kubelet major version parity with the Linux counterpart.
```

Comment:
What about other binaries like hybrid-overlay-node.exe?

Reply:
We will follow the same cadence with `hybrid-overlay-node.exe` also. I should have called that out here.